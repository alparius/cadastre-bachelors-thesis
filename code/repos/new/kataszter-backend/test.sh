#!/bin/bash

rm -f coverage.txt
echo 'mode: atomic' > coverage.txt

for d in $(go list ./... | grep -v testing); do
    go test -v -p 1 -race -covermode=atomic -coverprofile=coverage.out -coverpkg=$(go list ./...| grep -v testing | tr "\n" ",") $d
    if [ -f coverage.out ]; then
        tail -n +2 coverage.out >> coverage.txt || exit 255
        rm coverage.out
    fi
done
