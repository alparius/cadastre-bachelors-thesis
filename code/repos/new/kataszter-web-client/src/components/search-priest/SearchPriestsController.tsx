import React from "react";
import InputRange from "react-input-range";
import "react-input-range/lib/css/index.css";
import { FormattedMessage as FM, useIntl } from "react-intl";
import { Form, Header, Icon } from "semantic-ui-react";

import IPriestFilters, { filterEndYear, filterStartYear } from "../../data/priest/PriestFilters";
import BoldFM from "../../util/styledMessages/BoldFM";

const refreshButtonProps = {
    name: "refresh",
    basic: true,
    circular: true,
    compact: true,
    icon: true
};

interface ISearchPriestsControllerProps {
    chartPage?: boolean;
    filters: IPriestFilters;
    setFilters(newFilters: object): void;
    resetPagination(): void;
}

const SearchPriestsController: React.FC<ISearchPriestsControllerProps> = (props: ISearchPriestsControllerProps) => {
    const { chartPage, filters, setFilters, resetPagination } = props;
    const { formatMessage: fm } = useIntl();

    const handleChange = (event: any) => {
        const { name, value } = event.target;
        setFilters({ ...filters, [name]: value });
        resetPagination();
    };

    const handleRangeChange = (value: any, startLabel: string, endLabel: string) => {
        setFilters({ ...filters, [startLabel]: value.min.toString(), [endLabel]: value.max.toString() });
    };

    const resetRanges = (startLabel: string, endLabel: string) => {
        setFilters({ ...filters, [startLabel]: filterStartYear.toString(), [endLabel]: filterEndYear.toString() });
    };

    return (
        <Form>
            <Header as="h1" className="controllerHeader">
                {chartPage ? <FM id="charts.title" /> : <FM id="search.priests" />}
            </Header>
            <Form.Input id="name" name="name" label={fm({ id: "label.name" })} value={filters.name} onChange={handleChange} />
            <Form.Input id="birthTown" name="birthTown" label={fm({ id: "label.birthTown" })} value={filters.birthTown} onChange={handleChange} />
            <Form.Input id="placement" name="placement" label={fm({ id: "label.placement" })} value={filters.placement} onChange={handleChange} />
            <Form.Input
                id="area"
                name="area"
                label={fm({ id: "label.area" })}
                value={filters.area}
                onChange={handleChange}
                style={{ marginBottom: "0.5em" }}
            />

            <BoldFM style={{ fontSize: "13px" }} id="label.priest.activeYears" />
            <Form.Group style={{ marginTop: "0.5em", marginBottom: "2em" }} inline unstackable>
                <Form.Button onClick={() => resetRanges("activeStart", "activeEnd")} {...refreshButtonProps} width={2}>
                    <Icon name="refresh" />
                </Form.Button>
                <Form.Field width={1} />
                <Form.Field width={13}>
                    <InputRange
                        minValue={filterStartYear}
                        maxValue={filterEndYear}
                        onChange={(value: any) => handleRangeChange(value, "activeStart", "activeEnd")}
                        value={{ min: Number(filters.activeStart), max: Number(filters.activeEnd) }}
                    />
                </Form.Field>
            </Form.Group>

            <BoldFM style={{ fontSize: "13px" }} id="label.priest.birthYear" />
            <Form.Group style={{ marginTop: "0.5em", marginBottom: "2em" }} inline unstackable>
                <Form.Button onClick={() => resetRanges("birthStart", "birthEnd")} {...refreshButtonProps} width={2}>
                    <Icon name="refresh" />
                </Form.Button>
                <Form.Field width={1} />
                <Form.Field width={13}>
                    <InputRange
                        minValue={filterStartYear}
                        maxValue={filterEndYear}
                        onChange={(value: any) => handleRangeChange(value, "birthStart", "birthEnd")}
                        value={{ min: Number(filters.birthStart), max: Number(filters.birthEnd) }}
                    />
                </Form.Field>
            </Form.Group>

            <BoldFM style={{ fontSize: "13px" }} id="label.priest.deathYear" />
            <Form.Group style={{ marginTop: "0.5em", marginBottom: "2em" }} inline unstackable>
                <Form.Button onClick={() => resetRanges("deathStart", "deathEnd")} {...refreshButtonProps} width={2}>
                    <Icon name="refresh" />
                </Form.Button>
                <Form.Field width={1} />
                <Form.Field width={13}>
                    <InputRange
                        minValue={filterStartYear}
                        maxValue={filterEndYear}
                        onChange={(value: any) => handleRangeChange(value, "deathStart", "deathEnd")}
                        value={{ min: Number(filters.deathStart), max: Number(filters.deathEnd) }}
                    />
                </Form.Field>
            </Form.Group>

            <BoldFM style={{ fontSize: "13px" }} id="label.priest.subscriptionYear" />
            <Form.Group style={{ marginTop: "0.5em", marginBottom: "2em" }} inline unstackable>
                <Form.Button onClick={() => resetRanges("subscriptionStart", "subscriptionEnd")} {...refreshButtonProps} width={2}>
                    <Icon name="refresh" />
                </Form.Button>
                <Form.Field width={1} />
                <Form.Field width={13}>
                    <InputRange
                        minValue={filterStartYear}
                        maxValue={filterEndYear}
                        onChange={(value: any) => handleRangeChange(value, "subscriptionStart", "subscriptionEnd")}
                        value={{ min: Number(filters.subscriptionStart), max: Number(filters.subscriptionEnd) }}
                    />
                </Form.Field>
            </Form.Group>

            <BoldFM style={{ fontSize: "13px" }} id="label.priest.graduationYear" />
            <Form.Group style={{ marginTop: "0.5em", marginBottom: "2em" }} inline unstackable>
                <Form.Button onClick={() => resetRanges("graduationStart", "graduationEnd")} {...refreshButtonProps} width={2}>
                    <Icon name="refresh" />
                </Form.Button>
                <Form.Field width={1} />
                <Form.Field width={13}>
                    <InputRange
                        minValue={filterStartYear}
                        maxValue={filterEndYear}
                        onChange={(value: any) => handleRangeChange(value, "graduationStart", "graduationEnd")}
                        value={{ min: Number(filters.graduationStart), max: Number(filters.graduationEnd) }}
                    />
                </Form.Field>
            </Form.Group>
        </Form>
    );
};
export default SearchPriestsController;
