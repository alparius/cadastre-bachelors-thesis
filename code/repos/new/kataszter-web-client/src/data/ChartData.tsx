interface IChartData {
    key: any;
    data: any;
    counter: number;
}

export default IChartData;
