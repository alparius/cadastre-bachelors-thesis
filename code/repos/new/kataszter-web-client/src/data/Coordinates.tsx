interface ICoordinates {
    lat: number;
    lng: number;
}

export let NewCoordinates: ICoordinates = {
    lat: 0,
    lng: 0
};

export default ICoordinates;
