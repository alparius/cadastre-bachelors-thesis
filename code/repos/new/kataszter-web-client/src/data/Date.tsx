interface IDate {
    year: number;
    month: number;
    day: number;
    comment: string;
}

export let NewDate: IDate = {
    year: 0,
    month: 0,
    day: 0,
    comment: ""
};

export default IDate;
