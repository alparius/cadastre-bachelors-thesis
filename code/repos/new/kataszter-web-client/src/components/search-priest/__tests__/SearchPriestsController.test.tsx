/*===================
  SearchPriestsController.unit.test.tsx
 ===================*/
import React from "react";

import { NewTestPriestFilters } from "../../../util/test/data/test-data";
import { createComponentWithIntl, mountWithIntl } from "../../../util/test/test-wrapper";
import SearchPriestsController from "../SearchPriestsController";

const handlers = {
    setFilters: jest.fn(),
    resetPagination: jest.fn()
};

describe("SearchPriestsController element", () => {
    afterEach(() => {
        handlers.setFilters.mockClear();
        handlers.resetPagination.mockClear();
    });

    it("should render the form with search input fields", () => {
        const container = createComponentWithIntl(<SearchPriestsController filters={NewTestPriestFilters} {...handlers} />).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should call the setFilters function when the #name input field changes", () => {
        const container = mountWithIntl(<SearchPriestsController filters={NewTestPriestFilters} {...handlers} />);
        const input = container.find("input#name");
        input.simulate("change", { target: { value: "te" } });
        input.simulate("change", { target: { value: "st" } });
        expect(handlers.setFilters).toHaveBeenCalledTimes(2);
    });

    it("should call the setFilters function when the #birthTown input field changes", () => {
        const container = mountWithIntl(<SearchPriestsController filters={NewTestPriestFilters} {...handlers} />);
        const input = container.find("input#birthTown");
        input.simulate("change", { target: { value: "te" } });
        input.simulate("change", { target: { value: "st" } });
        expect(handlers.setFilters).toHaveBeenCalledTimes(2);
    });

    it("should call the setFilters function when the #placement input field changes", () => {
        const container = mountWithIntl(<SearchPriestsController filters={NewTestPriestFilters} {...handlers} />);
        const input = container.find("input#placement");
        input.simulate("change", { target: { value: "te" } });
        input.simulate("change", { target: { value: "st" } });
        expect(handlers.setFilters).toHaveBeenCalledTimes(2);
    });

    it("should call the setFilters function when the #area input field changes", () => {
        const container = mountWithIntl(<SearchPriestsController filters={NewTestPriestFilters} {...handlers} />);
        const input = container.find("input#area");
        input.simulate("change", { target: { value: "te" } });
        input.simulate("change", { target: { value: "st" } });
        expect(handlers.setFilters).toHaveBeenCalledTimes(2);
    });

    it("should call the resetPagination function when any input field changes", () => {
        const container = mountWithIntl(<SearchPriestsController filters={NewTestPriestFilters} {...handlers} />);
        let input = container.find("input#name");
        input.simulate("change", { target: { value: "te" } });
        input.simulate("change", { target: { value: "st" } });
        input = container.find("input#birthTown");
        input.simulate("change", { target: { value: "te" } });
        input.simulate("change", { target: { value: "st" } });
        input = container.find("input#placement");
        input.simulate("change", { target: { value: "te" } });
        input.simulate("change", { target: { value: "st" } });
        input = container.find("input#area");
        input.simulate("change", { target: { value: "te" } });
        input.simulate("change", { target: { value: "st" } });
        expect(handlers.resetPagination).toHaveBeenCalledTimes(8);
    });
});
