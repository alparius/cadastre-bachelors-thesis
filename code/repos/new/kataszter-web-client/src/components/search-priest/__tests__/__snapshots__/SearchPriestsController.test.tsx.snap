// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPriestsController element should render the form with search input fields 1`] = `
<form
  className="ui form"
  onSubmit={[Function]}
>
  <h1
    className="ui header controllerHeader"
  >
    Search priests
  </h1>
  <div
    className="field"
  >
    <label
      htmlFor="name"
    >
      Name
    </label>
    <div
      className="ui input"
    >
      <input
        id="name"
        name="name"
        onChange={[Function]}
        type="text"
        value="test"
      />
    </div>
  </div>
  <div
    className="field"
  >
    <label
      htmlFor="birthTown"
    >
      Town of birth
    </label>
    <div
      className="ui input"
    >
      <input
        id="birthTown"
        name="birthTown"
        onChange={[Function]}
        type="text"
        value="test"
      />
    </div>
  </div>
  <div
    className="field"
  >
    <label
      htmlFor="placement"
    >
      Placement
    </label>
    <div
      className="ui input"
    >
      <input
        id="placement"
        name="placement"
        onChange={[Function]}
        type="text"
        value="test"
      />
    </div>
  </div>
  <div
    className="field"
  >
    <label
      htmlFor="area"
    >
      Country / County / Diocese
    </label>
    <div
      className="ui input"
      style={
        Object {
          "marginBottom": "0.5em",
        }
      }
    >
      <input
        id="area"
        name="area"
        onChange={[Function]}
        type="text"
        value="test"
      />
    </div>
  </div>
  <b
    style={
      Object {
        "fontSize": "13px",
      }
    }
  >
    Active years
  </b>
  <div
    className="inline unstackable fields"
    style={
      Object {
        "marginBottom": "2em",
        "marginTop": "0.5em",
      }
    }
  >
    <div
      className="two wide field"
    >
      <button
        className="ui basic circular compact icon button"
        name="refresh"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="refresh icon"
          onClick={[Function]}
        />
      </button>
    </div>
    <div
      className="one wide field"
    />
    <div
      className="thirteen wide field"
    >
      <div
        aria-disabled={false}
        className="input-range"
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onTouchStart={[Function]}
      >
        <span
          className="input-range__label input-range__label--min"
        >
          <span
            className="input-range__label-container"
          >
            1800
          </span>
        </span>
        <div
          className="input-range__track input-range__track--background"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
        >
          <div
            className="input-range__track input-range__track--active"
            style={
              Object {
                "left": "0%",
                "width": "0%",
              }
            }
          />
          <span
            className="input-range__slider-container"
            style={
              Object {
                "left": "0%",
                "position": "absolute",
              }
            }
          >
            <span
              className="input-range__label input-range__label--value"
            >
              <span
                className="input-range__label-container"
              >
                NaN
              </span>
            </span>
            <div
              aria-valuemax={NaN}
              aria-valuemin={1800}
              aria-valuenow={NaN}
              className="input-range__slider"
              draggable="false"
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onTouchStart={[Function]}
              role="slider"
              tabIndex="0"
            />
          </span>
          <span
            className="input-range__slider-container"
            style={
              Object {
                "left": "0%",
                "position": "absolute",
              }
            }
          >
            <span
              className="input-range__label input-range__label--value"
            >
              <span
                className="input-range__label-container"
              >
                NaN
              </span>
            </span>
            <div
              aria-valuemax={2021}
              aria-valuemin={NaN}
              aria-valuenow={NaN}
              className="input-range__slider"
              draggable="false"
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onTouchStart={[Function]}
              role="slider"
              tabIndex="0"
            />
          </span>
        </div>
        <span
          className="input-range__label input-range__label--max"
        >
          <span
            className="input-range__label-container"
          >
            2021
          </span>
        </span>
      </div>
    </div>
  </div>
  <b
    style={
      Object {
        "fontSize": "13px",
      }
    }
  >
    Birthyear
  </b>
  <div
    className="inline unstackable fields"
    style={
      Object {
        "marginBottom": "2em",
        "marginTop": "0.5em",
      }
    }
  >
    <div
      className="two wide field"
    >
      <button
        className="ui basic circular compact icon button"
        name="refresh"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="refresh icon"
          onClick={[Function]}
        />
      </button>
    </div>
    <div
      className="one wide field"
    />
    <div
      className="thirteen wide field"
    >
      <div
        aria-disabled={false}
        className="input-range"
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onTouchStart={[Function]}
      >
        <span
          className="input-range__label input-range__label--min"
        >
          <span
            className="input-range__label-container"
          >
            1800
          </span>
        </span>
        <div
          className="input-range__track input-range__track--background"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
        >
          <div
            className="input-range__track input-range__track--active"
            style={
              Object {
                "left": "0%",
                "width": "0%",
              }
            }
          />
          <span
            className="input-range__slider-container"
            style={
              Object {
                "left": "0%",
                "position": "absolute",
              }
            }
          >
            <span
              className="input-range__label input-range__label--value"
            >
              <span
                className="input-range__label-container"
              >
                NaN
              </span>
            </span>
            <div
              aria-valuemax={NaN}
              aria-valuemin={1800}
              aria-valuenow={NaN}
              className="input-range__slider"
              draggable="false"
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onTouchStart={[Function]}
              role="slider"
              tabIndex="0"
            />
          </span>
          <span
            className="input-range__slider-container"
            style={
              Object {
                "left": "0%",
                "position": "absolute",
              }
            }
          >
            <span
              className="input-range__label input-range__label--value"
            >
              <span
                className="input-range__label-container"
              >
                NaN
              </span>
            </span>
            <div
              aria-valuemax={2021}
              aria-valuemin={NaN}
              aria-valuenow={NaN}
              className="input-range__slider"
              draggable="false"
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onTouchStart={[Function]}
              role="slider"
              tabIndex="0"
            />
          </span>
        </div>
        <span
          className="input-range__label input-range__label--max"
        >
          <span
            className="input-range__label-container"
          >
            2021
          </span>
        </span>
      </div>
    </div>
  </div>
  <b
    style={
      Object {
        "fontSize": "13px",
      }
    }
  >
    Death year
  </b>
  <div
    className="inline unstackable fields"
    style={
      Object {
        "marginBottom": "2em",
        "marginTop": "0.5em",
      }
    }
  >
    <div
      className="two wide field"
    >
      <button
        className="ui basic circular compact icon button"
        name="refresh"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="refresh icon"
          onClick={[Function]}
        />
      </button>
    </div>
    <div
      className="one wide field"
    />
    <div
      className="thirteen wide field"
    >
      <div
        aria-disabled={false}
        className="input-range"
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onTouchStart={[Function]}
      >
        <span
          className="input-range__label input-range__label--min"
        >
          <span
            className="input-range__label-container"
          >
            1800
          </span>
        </span>
        <div
          className="input-range__track input-range__track--background"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
        >
          <div
            className="input-range__track input-range__track--active"
            style={
              Object {
                "left": "0%",
                "width": "0%",
              }
            }
          />
          <span
            className="input-range__slider-container"
            style={
              Object {
                "left": "0%",
                "position": "absolute",
              }
            }
          >
            <span
              className="input-range__label input-range__label--value"
            >
              <span
                className="input-range__label-container"
              >
                NaN
              </span>
            </span>
            <div
              aria-valuemax={NaN}
              aria-valuemin={1800}
              aria-valuenow={NaN}
              className="input-range__slider"
              draggable="false"
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onTouchStart={[Function]}
              role="slider"
              tabIndex="0"
            />
          </span>
          <span
            className="input-range__slider-container"
            style={
              Object {
                "left": "0%",
                "position": "absolute",
              }
            }
          >
            <span
              className="input-range__label input-range__label--value"
            >
              <span
                className="input-range__label-container"
              >
                NaN
              </span>
            </span>
            <div
              aria-valuemax={2021}
              aria-valuemin={NaN}
              aria-valuenow={NaN}
              className="input-range__slider"
              draggable="false"
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onTouchStart={[Function]}
              role="slider"
              tabIndex="0"
            />
          </span>
        </div>
        <span
          className="input-range__label input-range__label--max"
        >
          <span
            className="input-range__label-container"
          >
            2021
          </span>
        </span>
      </div>
    </div>
  </div>
  <b
    style={
      Object {
        "fontSize": "13px",
      }
    }
  >
    Subscription year
  </b>
  <div
    className="inline unstackable fields"
    style={
      Object {
        "marginBottom": "2em",
        "marginTop": "0.5em",
      }
    }
  >
    <div
      className="two wide field"
    >
      <button
        className="ui basic circular compact icon button"
        name="refresh"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="refresh icon"
          onClick={[Function]}
        />
      </button>
    </div>
    <div
      className="one wide field"
    />
    <div
      className="thirteen wide field"
    >
      <div
        aria-disabled={false}
        className="input-range"
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onTouchStart={[Function]}
      >
        <span
          className="input-range__label input-range__label--min"
        >
          <span
            className="input-range__label-container"
          >
            1800
          </span>
        </span>
        <div
          className="input-range__track input-range__track--background"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
        >
          <div
            className="input-range__track input-range__track--active"
            style={
              Object {
                "left": "0%",
                "width": "0%",
              }
            }
          />
          <span
            className="input-range__slider-container"
            style={
              Object {
                "left": "0%",
                "position": "absolute",
              }
            }
          >
            <span
              className="input-range__label input-range__label--value"
            >
              <span
                className="input-range__label-container"
              >
                NaN
              </span>
            </span>
            <div
              aria-valuemax={NaN}
              aria-valuemin={1800}
              aria-valuenow={NaN}
              className="input-range__slider"
              draggable="false"
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onTouchStart={[Function]}
              role="slider"
              tabIndex="0"
            />
          </span>
          <span
            className="input-range__slider-container"
            style={
              Object {
                "left": "0%",
                "position": "absolute",
              }
            }
          >
            <span
              className="input-range__label input-range__label--value"
            >
              <span
                className="input-range__label-container"
              >
                NaN
              </span>
            </span>
            <div
              aria-valuemax={2021}
              aria-valuemin={NaN}
              aria-valuenow={NaN}
              className="input-range__slider"
              draggable="false"
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onTouchStart={[Function]}
              role="slider"
              tabIndex="0"
            />
          </span>
        </div>
        <span
          className="input-range__label input-range__label--max"
        >
          <span
            className="input-range__label-container"
          >
            2021
          </span>
        </span>
      </div>
    </div>
  </div>
  <b
    style={
      Object {
        "fontSize": "13px",
      }
    }
  >
    Graduation year
  </b>
  <div
    className="inline unstackable fields"
    style={
      Object {
        "marginBottom": "2em",
        "marginTop": "0.5em",
      }
    }
  >
    <div
      className="two wide field"
    >
      <button
        className="ui basic circular compact icon button"
        name="refresh"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="refresh icon"
          onClick={[Function]}
        />
      </button>
    </div>
    <div
      className="one wide field"
    />
    <div
      className="thirteen wide field"
    >
      <div
        aria-disabled={false}
        className="input-range"
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onTouchStart={[Function]}
      >
        <span
          className="input-range__label input-range__label--min"
        >
          <span
            className="input-range__label-container"
          >
            1800
          </span>
        </span>
        <div
          className="input-range__track input-range__track--background"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
        >
          <div
            className="input-range__track input-range__track--active"
            style={
              Object {
                "left": "0%",
                "width": "0%",
              }
            }
          />
          <span
            className="input-range__slider-container"
            style={
              Object {
                "left": "0%",
                "position": "absolute",
              }
            }
          >
            <span
              className="input-range__label input-range__label--value"
            >
              <span
                className="input-range__label-container"
              >
                NaN
              </span>
            </span>
            <div
              aria-valuemax={NaN}
              aria-valuemin={1800}
              aria-valuenow={NaN}
              className="input-range__slider"
              draggable="false"
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onTouchStart={[Function]}
              role="slider"
              tabIndex="0"
            />
          </span>
          <span
            className="input-range__slider-container"
            style={
              Object {
                "left": "0%",
                "position": "absolute",
              }
            }
          >
            <span
              className="input-range__label input-range__label--value"
            >
              <span
                className="input-range__label-container"
              >
                NaN
              </span>
            </span>
            <div
              aria-valuemax={2021}
              aria-valuemin={NaN}
              aria-valuenow={NaN}
              className="input-range__slider"
              draggable="false"
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onTouchStart={[Function]}
              role="slider"
              tabIndex="0"
            />
          </span>
        </div>
        <span
          className="input-range__label input-range__label--max"
        >
          <span
            className="input-range__label-container"
          >
            2021
          </span>
        </span>
      </div>
    </div>
  </div>
</form>
`;
